### PR TITLE
[SDK-2671] BrowserStack integration

### PR DESCRIPTION
**Pull request by stevehobbsdev**
This PR adds BrowserStack support for running E2E tests.

Summary of key changes:

* Replace Mochify, Zuul and Saucelabs with Karma, Browserify and BrowserStack
* Rename NPM test scripts so that they make more sense
* Updated development readme
* Update tests with some new utilities to make the tests more reliable. Not sure if they're completely solved yet but they seem to be reliable for now

### CODE REVIEW

File: test/enterprise_quick_auth.ui.test.js [hunk 1]
```diff
@@ -231,7 +231,12 @@ describe('enterprise quick auth', function() {
     });
 
     it('shows a login button with the windows icon', function() {
-      expect(h.hasQuickAuthButton(this.lock, 'windows', 'waad.com')).to.be.ok();
+      const _this = this;
+
+      setTimeout(
+        () => expect(h.hasQuickAuthButton(_this.lock, 'windows', 'waad.com')).to.be.ok(),
+        500
+      );
```
Comment (frederikprijck) [inline test/enterprise_quick_auth.ui.test.js:239]: Let's check if we can solve this without a timeout

File: test/layout.ui.test.js [hunk 2]
```diff
@@ -18,16 +18,21 @@ describe('layout', function() {
...
+        expect(h.hasLoginSignUpTabs(_this.lock)).to.be.ok();
+        expect(h.isLoginTabCurrent(_this.lock)).to.be.ok();
+        expect(h.hasSocialButtons(_this.lock)).to.be.ok();
+        expect(h.hasEmailInput(_this.lock)).to.be.ok();
+        expect(h.hasUsernameInput(_this.lock)).to.not.be.ok();
+        expect(h.hasPasswordInput(_this.lock)).to.be.ok();
+        expect(h.hasAlternativeLink(_this.lock)).to.be.ok(); // forgot password
+        expect(h.hasSubmitButton(_this.lock)).to.be.ok();
+        done();
+      }, 1000);
```
Comment (frederikprijck) [inline test/layout.ui.test.js:35]: Can we see if we can solve this without timeout

File: test/social_login.test.js [hunk 3]
```diff
@@ -21,15 +21,20 @@ describe('show lock connection scopes', function() {
...
+        expect(options.connection).to.be('facebook');
+        expect(options.connection_scope).to.be.an('array');
+        expect(options.connection_scope).to.have.length(2);
+        expect(options.connection_scope).to.contain('scope_1');
+        expect(options.connection_scope).to.contain('scope_2');
+        done();
+      });
+
+      h.clickSocialConnectionButton(_this.lock, 'facebook');
+    }, 500);
```
Comment (frederikprijck) [inline test/social_login.test.js:38]: Can we solve this without a timeout?

**Review by frederikprijck — CHANGES_REQUESTED** [review]

**Comment by stevehobbsdev** [comment]
@frederikprijck I've addressed the comments, I've also moved the BrowserStack tests into their own job.

**Review by frederikprijck — APPROVED** [review]

**Pull request merged**